includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-controller.php

```diff
@@ -40,21 +40,6 @@ public static function register_meta() {
 				'auth_callback'  => '__return_true',
 			]
 		);
-		\register_meta(
-			'post',
-			'mc_list_id',
```

Comment:
I didn't see this meta key being used anywhere in code, so I think we can get rid of it